minigrid/utils/bot.py

```diff
@@ -0,0 +1,1028 @@
+from __future__ import annotations
+
+import numpy as np
+
+from minigrid.core.world_object import WorldObj
+from minigrid.envs.babyai.core.verifier import (
+    AfterInstr,
+    AndInstr,
+    BeforeInstr,
+    GoToInstr,
+    ObjDesc,
+    OpenInstr,
+    PickupInstr,
+    PutNextInstr,
+)
+
+
+class DisappearedBoxError(Exception):
+    """
+    Error that's thrown when a box is opened.
+    We make the assumption that the bot cannot accomplish the mission when it happens.
+    """
+
+    def __init__(self, value):
+        self.value = value
+
+    def __str__(self):
+        return repr(self.value)
+
+
+def manhattan_distance(pos, target):
+    return np.abs(target[0] - pos[0]) + np.abs(target[1] - pos[1])
+
+
+class Subgoal:
+    """The base class for all possible Bot subgoals.
```

Comment:
The docstring are all of `NumPy` style I believe, pre-commit checks that docstrings are in `google` style. Could you update